### PR TITLE
more gc protection for coeffs, polys, and ideals

### PR DESCRIPTION
**Pull request by tthsqe12**
I got so sick of bugs that I just liberally applied the macro all over the place, possibly where it may not be needed.

### CODE REVIEW

File: src/ideal/ideal.jl [hunk 1]
```diff
@@ -124,13 +128,13 @@ coefficient ring were the rational numbers for example, the coefficients of the
 polynomials would be reduced to lowest terms.
 """
 function normalize!(I::sideal)
-   libSingular.id_Normalize(I.ptr, base_ring(I).ptr)
+   GC.@preserve I libSingular.id_Normalize(I.ptr, base_ring(I).ptr)
```
Comment (rfourquet) [inline src/ideal/ideal.jl:131]: Are we sure that `base_ring(I)` is preserved when `I` is preserved? (sorry if this is obvious, I don't have the structures in mind).

File: src/number/n_GF.jl [hunk 2]
```diff
@@ -125,9 +124,9 @@ end
 ###############################################################################
 
 function -(x::n_GF)
-    C = parent(x)
-    ptr = libSingular.n_Neg(x.ptr, C.ptr)
-    return C(ptr)
+   c = parent(x)
+   GC.@preserve x c p = libSingular.n_Neg(x.ptr, c.ptr)
```
Comment (rfourquet) [inline src/number/n_GF.jl:128]: I think it's a bit more visually clear to write here and elsewhere the `@preserve` within the `=` expression, i.e. 
```suggestion
   p = GC.@preserve x c libSingular.n_Neg(x.ptr, c.ptr)
```
But it's not at all a suggestion to update this PR, just for future ones.

**Comment by thofma** [comment]
Indentation is off at some places.

File: src/ideal/ideal.jl (hunk 1 above)
Comment (tthsqe12) [reply src/ideal/ideal.jl:131]: I think it is, but I will rewrite those places for extra security.

File: src/number/n_GF.jl (hunk 2 above)
Comment (tthsqe12) [reply src/number/n_GF.jl:128]: noted.

**Review by thofma — APPROVED** [review]

**Pull request merged**